etc/cvmfs/keys/computecanada.ca/soft.computecanada.ca.pub

```diff
@@ -0,0 +1,9 @@
+-----BEGIN PUBLIC KEY-----
```

Comment:
Is this key really unique to the "soft" repository only?  If not it would be better for it to have the more generic name.  If there are unique keys per repository then it's not a good idea to set CVMFS_KEYS_DIR because then all keys will be accepted for all repositories (in the future, when there is more than one)

Reply:
No, this is the shared key used for most of our repos.  It could have a generic name computecanada.ca.pub but some specificity is useful; we do think of it as "the soft key" for repos in "the soft stack". Otherwise we might end up with names like computecanada.ca.pub and computecanada2.ca.pub which are less descriptive.

We generally only use separate keys for repos that are hosted on separate servers.

Reply:
Ok but I think it is confusing to have it match the name of the repository.  cern.ch keys has some called `cern-it1.cern.ch.pub` etc, so maybe you could use `soft-repos.computecanada.ca.pub` or `soft-stack.computecanada.ca.pub`.  Or perhaps better yet, make a different subdirectory like `/etc/cvmfs/keys/computecanada.ca/softstack/computecanada.ca.pub` because that could make the directory usable for CVMFS_KEYS_DIR.  It's not good to put keys controlled by different people in the same CVMFS_KEYS_DIR directory since all of the keys will work for all of the repositories.  That's my suggestion.

Reply:
The keys are all controlled by the same people. The reason for having separate stacks (servers, keys) is just logistical, for example there is a dev one, and a data one with different infrastructure. They are all managed by the same team (us).

>  If there are unique keys per repository then it's not a good idea to set CVMFS_KEYS_DIR because then all keys will be accepted for all repositories

Isn't using multiple keys and multiple repos the only reason for using CVMFS_KEYS_DIR? It sounds like you are saying each keys dir should contain only a single key, which would be indistinguishable from using CVMFS_PUBLIC_KEY (though even that allows defining a list of multiple colon-separated files).  As long as nobody can inject fake keys into the config repo I'm not sure what the risk is.

Reply:
I think it's really for the case where multiple keys can sign any/all of the repositories.  That's the way CERN did it, and as far as I know so far they're the only ones using the feature.  Since they're all managed by the same team then it's probably ok to put them in the same directory, since you wouldn't care which key signed which repo.